test/unit/math/prim/prob/skew_double_exponential_test.cpp

```diff
@@ -0,0 +1,66 @@
+#include <stan/math/prim.hpp>
+#include <test/unit/math/prim/prob/vector_rng_test_helper.hpp>
+#include <test/unit/math/prim/prob/util.hpp>
+#include <gtest/gtest.h>
+#include <boost/math/distributions.hpp>
+#include <boost/random/mersenne_twister.hpp>
+#include <limits>
+#include <vector>
+
+class SkewDoubleExponentialTestRig : public VectorRNGTestRig {
+ public:
+  SkewDoubleExponentialTestRig()
+      : VectorRNGTestRig(10000, 10, {-2.5, -1.7, -0.1, 0.1, 2.0},
+                         {-3, -2, -1, 0, 2, 6}, {}, {}, {0.1, 1.0, 2.5, 4.0},
+                         {1, 2, 3, 4}, {-2.7, -1.5, -0.5, 0.0}, {-3, -2, -1, 0},
+                         {0.5}, {0}, {-0.1, 1.1}, {}) {}
+
+  template <typename T1, typename T2, typename T3, typename T_rng>
+  auto generate_samples(const T1& mu, const T2& sigma, const T3& tau,
+                        T_rng& rng) const {
+    return stan::math::skew_double_exponential_rng(mu, sigma, tau, rng);
+  }
+};
+
+TEST(ProbDistributionsSkewedDoubleExponential, errorCheck) {
+  check_dist_throws_all_types(SkewDoubleExponentialTestRig());
+}
+
+TEST(ProbDistributionsSkewedDoubleExponential, error_check) {
+  boost::random::mt19937 rng;
+  EXPECT_NO_THROW(stan::math::skew_double_exponential_rng(10.0, 2.0, .1, rng));
+
+  EXPECT_THROW(stan::math::skew_double_exponential_rng(10.0, 2.0, -1.0, rng),
+               std::domain_error);
+  EXPECT_THROW(stan::math::skew_double_exponential_rng(
+                   10, 2, stan::math::positive_infinity(), rng),
+               std::domain_error);
+}
+
+/*
+ * We test the skew double exponential rng by setting tau=0.5 which recovers
+ * a conventional double exponential distribution from Boost
+ */
+TEST(ProbDistributionsSkewedDoubleExponential, chiSquareGoodnessFitTest) {
+  boost::random::mt19937 rng;
+  int N = 10000;
+  int K = stan::math::round(2 * std::pow(N, 0.4));
+
+  std::vector<double> samples;
+  for (int i = 0; i < N; ++i) {
+    samples.push_back(
+        stan::math::skew_double_exponential_rng(2.0, 1.0, 0.5, rng));
+  }
+
+  // Generate quantiles from boost's double exponential distribution
+  boost::math::laplace_distribution<> dist(2.0, 1.0);
+  std::vector<double> quantiles;
+  for (int i = 1; i < K; ++i) {
+    double frac = static_cast<double>(i) / K;
+    quantiles.push_back(quantile(dist, frac));
```

Comment:
Oh dang, we are depending on Boost to compute quantiles. Okay so the quantiles are going to be inverse CDFs. Since the inverse CDF is easy, just code up the quantiles here manually.

The gotcha there is since we're using inverse CDF to do our sampling, if we had a bug in the inverse CDF here and in the rngs (we're writing them both and presumably copy-pasting), we still need to check the inverse CDFs here.

However, since we have a CDF function we trust, it's easy to check that the inverse CDF we write here is working.

We just do some variation of (in psuedo-code):

```
quantiles = c(0.05, 0.45, ...)
x = inverse_cdf(p);
EXPECT_FLOAT_EQ(cdf(x), p);
```

Does this make sense? We want to generated quantiles of the skew double exponential and then check that empirically our rng generates numbers from the right distribution (by computing how many samples land in these quantiles and doing a test to see if it's what we'd expect).